api/core/text2sql_sync.py

```
@@ -0,0 +1,635 @@
+    execution_time = time.perf_counter() - ctx.overall_start
+
+    # Save to memory in background if enabled
+    if ctx.memory_tool:
+        asyncio.create_task(
+            ctx.memory_tool.save_query_memory(
+                query=ctx.chat.queries_history[-1],
+                sql_query=final_sql,
+                success=True,
+                error=""
+            )
+        )
```

Ruff (0.14.14)

[warning] Store a reference to the return value of `asyncio.create_task`

(RUF006)